evm_loader/lib/src/types/deactivated_features.rs

```diff
@@ -0,0 +1,82 @@
+// use crate::tracing::tracers::state_diff::Account;
+use crate::rpc::Rpc;
+use once_cell::sync::Lazy;
+use solana_client::client_error::Result as ClientResult;
+use solana_sdk::pubkey::Pubkey;
+use std::collections::HashMap;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+
+static DEACTIVATED_FEATURES_UPDATE_SLOTS: u64 = 24 * 60 * 60; // approximately each 12h
+
+#[derive(Debug, Eq, PartialEq, Clone, Default)]
+struct DeactivatedFeaturesCache {
+    // feature to slot when activated, if not then None
+    pub data: HashMap<Pubkey, Option<u64>>,
+    pub last_slot: u64,
+}
+
+impl DeactivatedFeaturesCache {
+    fn should_update(&self, slot: u64) -> bool {
+        self.last_slot >= DEACTIVATED_FEATURES_UPDATE_SLOTS + slot
+    }
+}
+
+static DEACTIVATED_FEATURES: Lazy<Arc<Mutex<DeactivatedFeaturesCache>>> =
```

Comment:
Why don't you use `RwLock` instead of `Mutex`?

Reply:
Because we cannot call await of async function under lock of RwLock. In our case we need to call async Rpc::get_multiple_accounts